### PR TITLE
Always assume special urgency for contributor PR pings

### LINKED ISSUE #4327: Update the PR Review Reminder DAG with special timing for non-maintainers
## Problem

As identified in the Maintainer documentation [here](https://docs.openverse.org/general/contribution/good_first_and_help_wanted_issues.html#timing-and-process), we would like to keep the maintainer response time for reviewing community PRs as quick as possible. 3 days is expected to be reasonable.

## Description

Update the PR Review Reminder DAG to check if the PR author is not a maintainer and set the urgency to 3 days.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #4327 by @zackkrida

## Description

This PR modifies the PR reminder review DAG to use a special urgency level, `contributor` (3 business days), for any pull request that isn't opened by one of the `WordPress/openverse-maintainers` users (or a few other bot accounts). I modified the DAG workflow to use the TaskFlow API syntax, and updated tests accordingly as well.

3 days was chosen for this level based on the linked issue and this documentation: https://docs.openverse.org/general/contribution/good_first_and_help_wanted_issues.html#timing-and-process

## Testing Instructions

The new tests should pass and make sense. We don't have any PRs open right now that qualify for this new ping criteria, but if you have a copy of the GitHub PAT we use, you can test that the DAG still runs as expected with the new dependency.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py
Comment (dhruvkb) [inline catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py:21]: Nit: Can this be called "BOTS"?

File: catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py
Comment (dhruvkb) [inline catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py:32]: A perfect fit!

File: catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py
Comment (dhruvkb) [inline catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py:75]: If the set of maintainer logins cannot overlap with `NON_MAINTAINER_IGNORATIONS`, can we use `|` instead?

```suggestion
    } | NON_MAINTAINER_IGNORATIONS
```

**Review by dhruvkb — APPROVED** [review]
LGTM! I have some small nits but nothing so significant as to block merge.

File: catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py
Comment (AetherUnbound) [reply catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py:75]: Ugh, thanks for correcting this 😅 I knew there was an operator I wanted and started off with `&` (much to my incorrect dismay). This is the right one for sure!

**Review by obulat — APPROVED** [review]
Nice change!
Surprised at how many lines had to be changed for this.

**Comment by AetherUnbound** [comment]
> Nice change! Surprised at how many lines had to be changed for this.

Most of the lines were around machinery related to actually getting the maintainer members from GitHub - not something we had done before! 😄

**Pull request merged**
